src/utils/geometry_utils.py

```diff
@@ -0,0 +1,97 @@
+import numpy as np
+from pyquaternion import Quaternion
+
+
+def transform_matrix(translation: np.ndarray = np.array([0, 0, 0]),
+                     rotation: Quaternion = Quaternion([1, 0, 0, 0]),
+                     inverse: bool = False) -> np.ndarray:
+    """
+    Convert pose to transformation matrix.
+    :param translation: <np.float32: 3>. Translation in x, y, z.
+    :param rotation: Rotation in quaternions (w ri rj rk).
+    :param inverse: Whether to compute inverse transform matrix.
+    :return: <np.float32: 4, 4>. Transformation matrix.
+    """
+    tm = np.eye(4)
+
+    if inverse:
+        rot_inv = rotation.rotation_matrix.T
+        trans = np.transpose(-np.array(translation))
+        tm[:3, :3] = rot_inv
+        tm[:3, 3] = rot_inv.dot(trans)
+    else:
+        tm[:3, :3] = rotation.rotation_matrix
+        tm[:3, 3] = np.transpose(np.array(translation))
+
+    return tm
+
+
+def __corners(centers_xyz: np.ndarray,
+              sizes_lwh: np.ndarray,
+              orientation: Quaternion) -> np.ndarray:
+    """
+    Returns the bounding box corners.
+    :param wlh_factor: Multiply w, l, h by a factor to scale the box.
+    :return: <np.float: 3, 8>. First four corners are the ones facing forward.
+        The last four are the ones facing backwards.
+    """
+    l, w, h = sizes_lwh
+
+    # 3D bounding box corners. (Convention: x points forward, y to the left, z up.)
+    x_corners = l / 2 * np.array([1, 1, 1, 1, -1, -1, -1, -1])
+    y_corners = w / 2 * np.array([1, -1, -1, 1, 1, -1, -1, 1])
+    z_corners = h / 2 * np.array([1, 1, -1, -1, 1, 1, -1, -1])
+    corners = np.vstack((x_corners, y_corners, z_corners))
+
+    # Rotate
+    corners = np.dot(orientation.rotation_matrix, corners)
+
+    # Translate
+    x, y, z = centers_xyz
+    corners[0, :] = corners[0, :] + x
+    corners[1, :] = corners[1, :] + y
+    corners[2, :] = corners[2, :] + z
+
+    return corners
+
+
+def points_in_box(center_xyz: np.ndarray,

```

Comment:
i have the same method for once, but it's a bit different, maybe this can be named points_in_box_waymo?

Reply:
this method is unrelated to **waymo**: it checks whether the points are in the box specified by center and dimensions. Can  you use this method? I believe we should try to minimize the amount of geometry utils.

Reply:
it's very similar, could reuse it